kernel/thread.c

```diff
@@ -898,53 +1131,132 @@ size_t myst_get_num_threads(void)
     return _num_threads;
 }
 
+long myst_have_child_forked_processes(myst_thread_t* process)
+{
+    pid_t pid = process->pid;
+    myst_thread_t* p;
+    long ret = 0;
+
+    myst_spin_lock(&myst_process_list_lock);
+    p = process->main.prev_process_thread;
+
+    while (p)
+    {
+        if ((p->ppid == pid) && (p->clone.flags & CLONE_VFORK))
+        {
+            ret = 1;
+            break;
+        }
+        p = p->main.prev_process_thread;
+    }
+
+    if (!p)
+    {
```

Comment:
It would be nice to have an abstraction for operations applied to a process/thread list:

```c
bool predicate(myst_thread_t*);
void action(myst_thread_t*);
map2(myst_thread_t* current_thread, predicate, action);
```
Both myst_have_child_forked_processes and kill_child_fork_processes can be implemented on top of `map2`.

Reply:
hmm... not sure I like this type of approach

Reply:
It's a way to avoid repeating writing tedious and error-prone code to traverse a doubly linked list.